### ISSUE TITLE
R4 SDHC Dualcore​ compatibility

### ISSUE THREAD

**Issue by Endar-Marit** [kitling/flashcart_core#6]
Not sure what firmware you are using for the dualcore but I found after a bit of googling there was an unofficial YSMenu firmware listed for it on the woodr4 page as well as a firmware update .nds file, however for me the links are broken

Info from the [woodR4 Dualcore page](http://www.r4wood.com/products/R4-SDHC-Dual-Core-for-3DS-XL-LL-3DS-2DS-DSi.html)

Hope it helps, but for now im going to try to find that file

**Comment by Valeri0p** [comment]
Please provide photos of the card, CiphID, HW revision when trying to dump the flash with the tool and website. (Source: https://twitter.com/hedgeberg/status/896654926864158721)
@kitling maybe is better to create an issue template for better handing hw support request.

**Comment by EleventhSign** [comment]
I have this same card as @Endar-Marit . It's from www.r4isdhc.hk. It's detected as
`ChipID: 00000FC2`
` HW ID: 2A2A2A2A`

Since the website is down, using wayback machine, firmware update .nds file can be downloaded [here](https://web.archive.org/web/20170115092148/http://www.r4isdhc.hk/download_2014/3DSV7.0.0-13%20upgarde%20firmware.rar).

![dsc_0017](https://user-images.githubusercontent.com/30983865/29252280-9bdc62c0-8096-11e7-942e-bb1096188f6c.JPG)

Additional photos:

![dsc06067](https://user-images.githubusercontent.com/30983865/29252300-d564dc48-8096-11e7-9396-bfcd82b59f1e.JPG)

![dsc06045](https://user-images.githubusercontent.com/30983865/29252289-af1b46c6-8096-11e7-9676-bfb648fece76.JPG)

![dsc06053](https://user-images.githubusercontent.com/30983865/29252294-bcb8dbe0-8096-11e7-89e3-fcee55acafe4.JPG)

**Comment by NicknameGoesHere** [comment]
The R4 SDHC Dual-Core is not supported yet (I think so?)

**Comment by Valeri0p** [comment]
Not for now, it's on the todo list... thanks for the report anyway!
I've realized that my dead r4 (pretty similar to [this other](https://github.com/kitling/flashcart_core/issues/3#issuecomment-322060996)) it's an SDHC clone! :D

**Comment by Legarad** [comment]
I have an R4 DS SDCH Dual Core 2013 
Website www.r4isdch.com
ChipID:24242FE6
HW Rev:E4E4E4E4

**Comment by Valeri0p** [comment]
@Legarad that seems totally another flashcard, and the website is pretty dead. Please open another issue providing, other than that, photos of the card and backups of the firmware and updater if you can.

**Comment by Legarad** [comment]
@Valeri0p Sorry, it was a writing error because I was in a hurry to get to my work, the actual website is www.r4isdhc.com. It's a discontinued flashcart from that page

**Comment by Legarad** [comment]
@Valeri0p Update the firmware via webarchive and now the R4 SDHC has the information of @EleventhSign  's R4

**Comment by 607jf** [comment]
![imgp0144](https://user-images.githubusercontent.com/31044173/29327293-833c9e0c-81a3-11e7-8ada-86ae2b1ff075.JPG)

![imgp0133](https://user-images.githubusercontent.com/31044173/29327277-7c327d52-81a3-11e7-84e1-53cdc150f4d0.JPG)

![imgp0141](https://user-images.githubusercontent.com/31044173/29327268-7228c2da-81a3-11e7-9532-86ea8351a237.JPG)
R4 DS SDCH Dual Core 2016

www.r4isdhc.com

ChipID:000007C2
HW Rev:FFFFFFFF

![imgp0140](https://user-images.githubusercontent.com/31044173/29327255-6a3538f6-81a3-11e7-999e-8a497ad6ea0b.JPG)

**Comment by EleventhSign** [comment]
@Legarad did you say that you updated your cart using the updater I linked? If so, can you try to detect it again with the flasher and post the pic here?

_Sent from my Sony F5321 using [FastHub](https://play.google.com/store/apps/details?id=com.fastaccess.github)_

**Comment by Legarad** [comment]
@EleventhSign No, I use the files of this website
https://web.archive.org/web/20131106221949/http://www.r4isdhc.com:80/download_white_new2013/

**Comment by EleventhSign** [comment]
You could run an md5 check on both .nds files to determine if they are identical. Nevertheless, if the pcb of your cart isn't the one as I posted here, please open as a new issue.

**Comment by Legarad** [comment]
@EleventhSign your link working to me.

**Comment by DiamondMinerITA** [comment]
any news?

**Comment by Zemogiter** [comment]
I have this card:
http://www.r4isdhc.com/wp-content/themes/nds9/images/w/31.jpg
ChipID: 07CD3A86
HW REV: 6832C803
firmware: 
http://www.r4isdhc.com/KerneDownload/R4i-V3.9b%20English.zip

![20170818_231411](https://user-images.githubusercontent.com/24385408/29478076-6d1301fa-846b-11e7-8a8e-6ded9fb2c3e8.jpg)


![20170818_231444](https://user-images.githubusercontent.com/24385408/29478094-8ab64050-846b-11e7-8f1b-6c275be46f74.jpg)

**Comment by 1vanchom** [comment]
I have a question...
I have an r4 sdhc dualcore 2014 from r4isdhc.hk
inside is similar at @EleventhSign 's flashcard from like 10 posts above
the issue is that the flashcard is bricked... tried the swapping metod with dslite... and nothing... 
it's recogniced with:
ChipID: 00000FC2
HW Rev: 2A2A2A2A
same chipID and hw rev... but flashcard is bricked...
if there will be a ntrboot release with dualcore support... do you think it may work? is there a way to unbrick it?

**Comment by MilkyPL** [comment]
I have r4isdhc.com dual core (2017 version).
ChipID: 1D5C137F
HW Rev: 137EA6B5
This is original cart, not from weird chineese sites.

_Sent from my Htc HTC Desire 820 using [FastHub](https://play.google.com/store/apps/details?id=com.fastaccess.github)_

**Comment by Juanito9001** [comment]
@MilkyPL i have the same card but the chipID abd the HWrev is different, i saw the chips of my card an they are originals, ( like in the page) and the kernel works, and works with N3ds 11.5, but i dont know why your card and my cards appears differents in chipID and HWrev, any idea?

_Sent from my Samsung SM-G610M using [FastHub](https://play.google.com/store/apps/details?id=com.fastaccess.github)_

**Comment by MilkyPL** [comment]
@Juanito9001 I thought it was weird too, do you have the 2017 version?

_Sent from my Htc HTC Desire 820 using [FastHub](https://play.google.com/store/apps/details?id=com.fastaccess.github)_

**Comment by Zemogiter** [comment]
@MilkyPL @Juanito9001 every time I try the exploit it gives me diffirent HV Rev and ChipID numbers.

**Comment by Juanito9001** [comment]
@MilkyPL yes, i hace the r4 dual core and the r4 silver rts lite, and both are the same chipID and HWrev
000007C2
FFFFFFFF

_Sent from my Samsung SM-G610M using [FastHub](https://play.google.com/store/apps/details?id=com.fastaccess.github)_

**Comment by NicknameGoesHere** [comment]
I have almost the exact same board as Zemogiter's Dual Core, but I have a different card (R4 Gold Pro, same site) but the chip ID and Hw Rev are completely different.
4133AAB0
F6FF2DAC

**Comment by Bl1tzkrieg** [comment]
The board of my card is very similar to this one, but is a R4i Unlimited Upgradable, the website on the sticker is r4i-sdhc.kh instead of r4isdhc.hk, the chip ID is the sabe BUT the hw rev is 00000000.

**Comment by Jerry-Shaw** [comment]
@1vanchom , same here, bricked R4i, same chipID & hw rev, wait for some tools to read from a good one and write back to the bricked one.

**Comment by F3bruary1** [comment]
Legarad, DualCore 2013 = Clone of R4 Gold (Wood) 
Firmware and kernel https://yadi.sk/d/oMCYxwU-kUfvh

**Comment by Altairseven** [comment]
dual core 2014:
chip id and hw revision same as @MilkyPL (000007C2, FFFFFFFF)

**Comment by aocodes** [comment]
dual core 2016 (www.r4isdhc.hk)
chip id: 2F2F2F2F
hw rev: 80808080

**Comment by SaikoPikachuu** [comment]
got r4 sdhc dual core 2016 chip id 0000072C hw rev FFFFFFFF someonepls make it wor

**Comment by 607jf** [comment]
ChipID:000007C2
HW Rev:FFFFFFFF
![imgp0133](https://user-images.githubusercontent.com/31044173/30168778-3f65e8e2-939f-11e7-9259-f5d52429a5f3.JPG)
![imgp0140](https://user-images.githubusercontent.com/31044173/30168780-3fa1bf98-939f-11e7-8a24-c504d40d12c1.JPG)
![imgp0141](https://user-images.githubusercontent.com/31044173/30168777-3f655792-939f-11e7-8e32-844ec56cf260.JPG)
![imgp0144](https://user-images.githubusercontent.com/31044173/30168779-3f6d175c-939f-11e7-8b8b-da20cdca95a8.JPG)

**Comment by Juanito9001** [comment]
@607jf, your flashcart was mentioned before in other issue, this issue is especifically for R4isdhc.hk flashcart

your flashcart is from r4isdhc.com, search the issue and post your card there, btw, erase your flashcart from this issue because is not the same

**Comment by DiamondMinerITA** [comment]
This is pretty much the r4igold.cc one. look at #3 .

**Comment by ultraviolet150** [comment]
Just my 2 cents; it may shed some light into which factory these (and others shipped from)

In browsing web archive, i realised that the contact email is the same for www.r4isdhc.hk, www.r4itt.net, r4i-gold.me and www.r4ids.cc. (most of these no longer work - some do).

Its interesting to see that the firmware for the SDHC Dual Core at:
http://web.archive.org/web/20140328054251/http://www.r4isdhc.hk/download_2014/3DSV7.0.0-13%20upgarde%20firmware.rar

and the firmware for the R4I-GOLD PRO at:
http://r4i-gold.me/en/download/skin/2014/3DSV7.0.0-13%20upgarde%20firmware.rar

See to be exactly the same?

like I said... Just my 2 cents.

**Comment by chealp** [comment]
I got R4 sdhc Dual-Core r4isdhc.hk 2017 with firmware version 7.07. It was working with wood kernel _DS_MENU.DAT. when i copied another kernel with R4.DAT It got black screen with error to turn off dsi. then when i copied back the wood kernel the problem still persist. I saw the swap trick method(I have 2 of the cart) but I cant continue because the version of the firmware file is 7.00 which is less than. Is there any way to bypass version check? or a way to write the firmware a different way?

**Comment by Jerry-Shaw** [comment]
@chealp We met the same problem. Tried, but failed, even with ejecting the good card when the flash program was ready to check the version just in seconds. I got version of known, but the program blocked on erasing device on NDSL, or, all seemed good to go on 3DS, but card is still bricked.

**Comment by JellyVGC** [comment]
how would i go about finding out my chip id and HW Revision?

**Comment by kitling** [comment]
As @DiamondMinerITA said, this is basicly the r4igold.cc. Updaters seem to use the same commands, too.

The incomplete r4isdhc dualcore support in mainline was for this cart.

So, does anyone here have a logic analyzer or a beaglebone (for https://github.com/abhishek-kakkar/BeagleLogic)? I'll go ask on the r4igold.cc issue as well.

**Comment by zoogie** [comment]
Here's my 2017 model of this card. PCB looks different.

![2017-10-28 20 11 49](https://user-images.githubusercontent.com/28328903/32139918-7626a6cc-bc1d-11e7-9acc-3fea0caf7ed7.jpg)

**Comment by zoogie** [comment]
Ok, it turns out this .hk card of mine is actually the .com.cn card. Go figure, lol.

**Comment by yva96** [comment]
Another Card It's from www.r4isdhc.in . It's detected as 
ChipID: 00000FC2
HW ID: 2A2A2A2A
Reference here 
#103

**Comment by turntwo-gh** [comment]
I have this cart and a rudimentary logic analyzer with a Rasp-Pi, I can likely get 👀 on this if there is some specifics of what needs to be looked at.

**Comment by kitling** [comment]
@turntwo-gh well shoot. What is its sampling rate? You'll need something that can measure a protocol running at 4-6 Mhz. (And if it can, could you point me towards the software you're using? Probably means I was doing something wrong with [NTRPi](https://github.com/kitling/NTRPi/))

Regardless: You'll want a way to break out the cart signals in order to have access to them while your console is communicating with it. (I'm using a Action Replay with everything removed so that it just passes through the signals and has an additional way of breaking out the signals) You'll want at least D0-D7 and CLK, though it'd probably be preferable if you had CS1, CS2, and RST as well. ([pinout](http://www.hardwarebook.info/DS_Game_Card))

@angelsl wrote a [protocol decoder](https://github.com/angelsl/nds-pmla) for sigrok, but the gist of the protocol is: if CS1 is low, then it's a bidirectional parallel protocol that eventually transitions to use encryption (which the protocol decoder attempts to decode). 8 bytes are sent as a command, with everything afterwards until CS1 goes high again being a response. We're looking for a command that is not part of what has been [normally observed](http://problemkaputt.de/gbatek.htm#dscartridgeprotocol) that is used to tell the cart to respond to other flashcart commands, probably sent while it is in key2 mode.

If you have other questions, or would like to talk in more real-time, please come to the IRC channel where we hang out, #Cakey on freenode. I believe there's a webchat link somewhere in the README if you're not familiar with IRC.

**Comment by turntwo-gh** [comment]
😨 currently, not nearly enough channels and sampling frequency. Closest I've seen is [this](http://tuxbabe.eu/raspalyzer.html) that sample up to 3MHz and runs on bare-metal. Will try circling back on this once I get better hardware, you provided solid info to go on though, thanks.

**Comment by kitling** [comment]
Hah. honestly, I'm not that surprised. Actually, I am a bit surprised that that's only the speed they've got, but not very. I didn't expect much higher.

I believe there are some cheap cheap logic analyzers that are compatible w/ sigrok, @angelsl used two and combined their outputs for their setup. Again, feel free to come to IRC to ask questions.

**Comment by angelsl** [comment]
My setup didn't manage to capture the faster clock rate though. Or I didn't get something right (?).

I may take a look at the raw capture again.

**Comment by joasnake** [comment]
Hi , i have 2 cards aka r4sdhc dual core 2014 model from r4isdhc.hk do these support the ntrboot method ?

i still use them to play the ds games on my older nintendo ds and nintendo ds lite but bought a new 3ds xl today, wanted to check

**Comment by yva96** [comment]
hi i have 2 Cards aka r4sdhc dual core 2015 model from r4isdhc.in 
comes up as spongebob Squarepantis 

ntrboot flasher says flashcard not supported 
ChipID: 00000FC2
HW Rev: 2A2A2A2A
![r4i_r2_c42014new](https://user-images.githubusercontent.com/33157997/35520551-a06b5732-0517-11e8-956a-842e2dcd2a0d.jpg)

**Comment by cannedjava** [comment]
Hey guys, I know all of you are busy and all, but could I ask if there are plans to support this card, or if there's too much of a hassle to support it in favor of other cards? I'm asking because this is the only card that I could find locally, and ordering online really isn't in the cards for me.
